Assets/Scripts/StaminaBar.cs

```diff
@@ -70,4 +87,43 @@ void UpdateFillAmount(float newFill)
 
         OnChange?.Invoke(newFill);
     }
+
+    //Fading in and out
+    public void FadeOut()
+    {
+        // full color --> invisible
+        StartCoroutine(FadeCoroutine(1f, 0f, staminaFillImage));
+        StartCoroutine(FadeCoroutine(1f, 0f, staminaBarImage));
+        StartCoroutine(FadeCoroutine(1f, 0f, greyOutlineImage));
+    }
+
+    public void FadeIn()
+    {
+        print("fading out");
+        // invisible --> full color
+        StartCoroutine(FadeCoroutine(0f, 1f, staminaFillImage));
+        StartCoroutine(FadeCoroutine(0f, 1f, staminaBarImage));
+        StartCoroutine(FadeCoroutine(0f, 1f, greyOutlineImage));
+    }
+
+    private IEnumerator FadeCoroutine(float startAlpha, float endAlpha, Image image)
+    {
+        float currentAlpha = startAlpha;
+        while (Mathf.Abs(currentAlpha - startAlpha) < Mathf.Abs(startAlpha - endAlpha))
+        {
+            image.color = new Color(image.color.r, image.color.g, image.color.b, currentAlpha);
+            currentAlpha += fadeSpeed * Time.deltaTime * Mathf.Sign(endAlpha - startAlpha);
+            yield return null;
+        }
+

```

Comment:
Force at the end to reach the endAlpha or else it will sometimes not go all the way:
```suggestion

image.color = new Color(image.color.r, image.color.g, image.color.b, endAlpha);
```

![image](https://user-images.githubusercontent.com/11907795/75378630-0a5c7880-58a2-11ea-8d93-0c54de619ff6.png)